workspace.sh

```diff
@@ -186,6 +186,7 @@ workspace() {
     case "${1:-}" in
     add)
         shift
+         [ -z "${1:-}" ] && { show_help; return 1; }
```

Comment:
This idiom should be inverted. If the `[ -z ... ]` condition fails, the whole program will exit at that point due to running with `set -e`. The correct version here should be `[ -n "${1:-}" ] || { show_help; return 1; }`.